### PR TITLE
List team members by role

### LINKED ISSUE #926: add team.listMembersWithRole(...) to get which team members are maintainer vs member
**Describe the bug**
org has `listMembersWithRole`, team does not. Making it impossible to check if a team member is maintainer or member.

Suggestion: add listMembersWithRole to team just like org has it.

### PR DESCRIPTION
**Pull request by tginiotis-at-work**
# Description 
Upgrades the listing method to also take a role.
Closes #926.

# Before submitting a PR:
We love getting PRs, but we hate asking people for the same basic changes every time. 

- [x] Push your changes to a branch other than `master`. Create your PR from that branch.    
- [x] Add JavaDocs and other comments
- [x] Write tests that run and pass in CI. See [CONTRIBUTING.md](CONTRIBUTING.md) for details on how to capture snapshot data.
- [x] Run `mvn clean compile` locally. This may reformat your code, commit those changes.
- [x] Run `mvn -D enable-ci clean install site` locally. If this command doesn't succeed, your change will not pass CI.

# When creating a PR: 

- [x] Fill in the "Description" above. 
- [x] Enable "Allow edits from maintainers".

### CODE REVIEW

File: src/main/java/org/kohsuke/github/GHTeam.java [hunk 1]
```diff
@@ -166,12 +179,18 @@ public GHDiscussion getDiscussion(long discussionNumber) throws IOException {
...
      *
+     * @param role
+     *            the role to filter the result by
+     *
      * @return the paged iterable
      * @throws IOException
      *             the io exception
      */
-    public PagedIterable<GHUser> listMembers() throws IOException {
-        return root.createRequest().withUrlPath(api("/members")).toIterable(GHUser[].class, item -> item.wrapUp(root));
```
Comment (bitwiseman) [inline src/main/java/org/kohsuke/github/GHTeam.java:174]: This is an API change. Please retain the existing public method.

File: src/main/java/org/kohsuke/github/GHTeam.java [hunk 2]
```diff
@@ -145,6 +145,19 @@ public void setPrivacy(Privacy privacy) throws IOException {
...
+    /**
+     * List members with specified role paged iterable.
+     *
+     * @param role
+     *            the role
+     * @return the paged iterable
+     * @throws IOException
+     *             the io exception
+     */
+    public PagedIterable<GHUser> listMembersWithRole(String role) throws IOException {
```
Comment (bitwiseman) [inline src/main/java/org/kohsuke/github/GHTeam.java:157]:
```suggestion
    public PagedIterable<GHUser> listMembers(String role) throws IOException {
```

**Review by bitwiseman — CHANGES_REQUESTED** [review]

File: src/main/java/org/kohsuke/github/GHTeam.java (hunk 2 above)
Comment (tginiotis-at-work) [reply src/main/java/org/kohsuke/github/GHTeam.java]: I have renamed the method.

File: src/main/java/org/kohsuke/github/GHTeam.java (hunk 1 above)
Comment (tginiotis-at-work) [reply src/main/java/org/kohsuke/github/GHTeam.java]: I have readded the original public method.

**Review by bitwiseman — APPROVED** [review]

**Pull request merged**